### PR TITLE
Add docs for user portal theming

### PR DESCRIPTION
**Pull request by emswbandara**
This PR contains docs for following.

- Customizing user portal UI.
- Setting up User Portal in a development environment.

### CODE REVIEW

File: en/docs/develop/customizing-user-portal-ui.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+    To change the title, in **index.jsp** file of the web app, change the `<title>` tag.
+    ```html
+        <title>Night Owl Express</title>
+    ```
+
+### Step 3: Deploying the changes in the web app
+
+We are at the final step of the process which is the deployment. Follow the sequence of steps listed below to deploy
+the changes performed in the previous steps.
```
Comment (sherenewso2) [inline en/docs/develop/customizing-user-portal-ui.md:188]: The final step of the process is the deployment. Follow the....

File: en/docs/develop/customizing-user-portal-ui.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+1.  Build the theme module.
+```java
+    # from inside modules/theme
+    $npm run build
+```
+2.  Copy the artifacts to the web app.
+
+    The built artifacts will be available inside the **lib** folder. Copy everything to the clipboard and navigate to
```
Comment (sherenewso2) [inline en/docs/develop/customizing-user-portal-ui.md:197]: **lib** --> `lib`

Can we give the full path to the lib folder here?

File: en/docs/develop/customizing-user-portal-ui.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+     the **user-portal** web app in the IS pack.
+    user-portal web app is available in the location `<IS_HOME>/repository/deployment/server/webapps/user-portal`
+
+    Replace the contents inside `/libs/styles/css` with the copied artifacts.
```
Comment (sherenewso2) [inline en/docs/develop/customizing-user-portal-ui.md:201]: Copy everything to the clipboard and paste it inside the `<IS_HOME>/repository/deployment/server/webapps/user-portal/libs/styles/css` folder.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# Setting-Up User Portal in a Development Environment
+
+In [WSO2 Identity Server 5.10.0](https://wso2.com/identity-and-access-management/) the user portal is available under
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:3]: Let's remove the version number from WSO2 Identity Server

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+ the URL `http(s)://<SERVER_HOST>/t/<TENANT_DOMAIN>/user-portal`. If you are planning on setting up the repo in a
+ development environment follow the steps given below.
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:5]: Follow the steps below to set up the repo in a development environment.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+!!! note "Before you begin"
+
+    1. Install [node](https://nodejs.org/en/download/) if you haven’t already(npm is already bundled with node).
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:9]: haven't --> have not

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    2. Install [maven](https://maven.apache.org/download.cgi) (Needed to run mvn commands).
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:10]: (Needed to run mvn commands) --> (this is needed to run `mvn` commands)

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    3. A running Identity Server v 5.10. (If you want to build from source follow the instructions listed [here]
+    (https://github.com/wso2/product-is)).
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:12]: A running instance of WSO2 Identity Server (to build from source, follow the instructions given [here]()

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+
+### Step 1: Configuring Identity Server
+
+!!! tip
+    Default dev origin, hostname and port set for the User Portal is https://localhost:9000. So the following
+    configurations will have to change accordingly if you change the port or any other configurations.
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:19]: Note that the default dev origin, hostname, and port that is set for the user portal is `https://localhost:9000`. Change the following configurations accordingly if you have changed this port or any other configuration.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+
+1.  Allowing Cross Origin requests to the Identity Server.
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:22]: Enable cross-origin requests for WSO2 Identity Server.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+    Cross origin requests will be blocked by default as a security measure. So you have to put the following CORS
+    filter to the config file `<IS_HOME>/repository/resources/conf/templates/repository/conf/tomcat/web.xml.j2` in
+    WSO2 Identity Server distribution pack.
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:26]: Cross-origin requests are blocked by default in WSO2 IS as a security measure. Add the following CORS filter to the `<IS_HOME>/repository/resources/conf/templates/repository/conf/tomcat/web.xml.j2` file to enable it.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md (hunk 8 above)
Comment (sherenewso2) [reply en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:12]: can we just give a link to the IS binary.zip instead?

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+        <filter-mapping>
+            <filter-name>CORS</filter-name>
+            <url-pattern>/*</url-pattern>
+            <dispatcher>REQUEST</dispatcher>
+            <dispatcher>FORWARD</dispatcher>
+        </filter-mapping>
+    ```
+2.  Add your hostname and port as a trusted FIDO2 origin.
+
+    Whitelist the dev URL in the FIDO configurations found in the file
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:53]: 2. Whitelist your hostname and port as a trusted FIDO2 origin.

     Add the dev url to the `<Origin>` tag of the FIDO configurations in the `<IS_HOME>/repository/resources/conf/templates/repository/conf/identity/identity.xml.j2` file.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+            <FIDO2TrustedOrigins>
+                    {% for origin in fido.trusted.origins %}
+                    <Origin>{{origin}}</Origin>
+                    {% endfor %}
+                    <Origin>https://localhost:9000</Origin>
+            </FIDO2TrustedOrigins>
+        </FIDO>
+    ```
+
+3.  Restart the Identity Server.
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:69]: the Identity Server --> WSO2 Identity Server

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+    Log in to the management console `https://<SERVER_HOST>/carbon`
+    Go to service provider listing and click on edit in **User Portal** list item.
+
+    In the edit view, expand the **Inbound Authentication Configuration** section, and further expand the **OAuth/OpenID
+     Connect Configuration**. Click on edit under User Portal list item.
+
+    You can simple change the port to 9000 or you can add a regexp as follows.
+    ```
+        regexp=(https://localhost:9443/user-portal/login|https://localhost:9000/user-portal/login)
+    ```
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:81]: Number these as steps indented under step 4. 

1. Log in to the WSO2 IS management console. 

2. Click **Service Providers > List**.

3. Click **Edit** to edit the **User Portal** service provider. 

4. Expand **Inbound Authentication Configuration**, and then expand **OAuth/OpenID Connect Configuration**. Click **Edit**. 

5. Change the **Callback URL** field to reflect the port as 9000 or you can add a regexp as follows.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+
+### Step 2: Building the Identity Apps repository
+
+You can fork the original repo or directly clone the original repo and start working on it, but it is recommended to
+create your own fork.
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:86]: Fork the original repository to start working on it. You can also directly clone the original repo but it is recommended to create your own fork.

File: en/docs/develop/setting-up-user-portal-in-a-dev-environment.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+```java
+    git clone https://github.com/brionmario/identity-apps
+    cd identity-apps
+    mvn clean install or npm run build
+```
+
+### Step 3: Running the User Portal in dev mode
```
Comment (sherenewso2) [inline en/docs/develop/setting-up-user-portal-in-a-dev-environment.md:93]: User Portal --> user portal

File: en/docs/develop/customizing-user-portal-ui.md [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+    ![default-user-portal-theme](../assets/img/develop/default-user-portal-theme.png)
+
+    Check the [Semantic UI documentation](https://semantic-ui.com/usage/theming.html) if you wish to learn more about
+     Semantic UI theming.
+
+!!! note "Before you begin"
+
+    1. Check out the corresponding identity apps source code from the [repo](https://github.com/wso2/identity-apps)
+    and set it up in the [development environment]((../../develop/setting-up-user-portal-in-a-dev-environment)). You
+    can check out the `v1.0.72` tag since `v1.0.72` of identity-apps was used in IS 5.10.0
```
[the remark this answers is not in the document]
Comment (emswbandara) [reply en/docs/develop/customizing-user-portal-ui.md:26]: @brionmario would it be ok if we point to the latest tag here? would it cause problems?

[the remark this answers is not in the document]
Comment (emswbandara) [reply en/docs/develop/customizing-user-portal-ui.md:26]: here it needs to be exact tag for the version.. otherwise it would break the app

**Pull request merged**